### ISSUE TITLE
TypeORM stores incorrect time to db when column data type is 'time' and database is SQL Server

### ISSUE THREAD

**Issue by penttin5** [typeorm/typeorm#10402]
### Issue description

TypeORM stores incorrect time to db when column data type is 'time' and database is SQL Server

### Expected Behavior

After calling `repository.save({ timeString: '08:00:00' })`, the db should contain time '08:00:00'

### Actual Behavior

If useUTC is true, calling `repository.save({ timeString: '08:00:00' })` results in incorrect time. Sometimes the result is '05:00:00' and sometimes '05:00:01'.
My guess is that TypeORM transforms the time string to Javascript Date with local timezone (my timezone is UTC+3, so 08:00:00+03:00), then converts it into UTC and stores the UTC time to db. I guess that TypeOrm gets the milliseconds from current instant and rounds them to the nearest second, so the resulting seconds are correct for about 50% of the inserts/updates.

If useUTC is false, the result is otherwise correct, but the seconds are off by one every now and then.

### Steps to reproduce

1. set useUTC=true or false in TypeORM's options
2. define a time column to entity
   ```ts
   @Entity()
   export class TimeTestEntity {
     @Column({ type: 'time' })
     timeString!: string
   ```
3. save time value into the db 
   ```ts
    const entity = {
      timeString: '08:00:00',
    }
    await dataSource.getRepository(TimeTestEntity).save(entity)
   ```
4. value stored in the db is either '05:00:00' or '05:00:01' if useUTC is true and '08:00:00' or '08:00:01' if useUTC is false

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |   macOs       |
| Node.js version     | 16.14.0  |
| Typescript version  | 5.2.2 |
| TypeORM version     | 0.3.17  |
| mssql |  9.3.0 |

### Additional Context

We could circumvent the bug by adjusting the time values in ValueTransformer, but https://github.com/typeorm/typeorm/issues/10397 prevents this.

### Relevant Database Driver(s)

- [ ] aurora-mysql
- [ ] aurora-postgres
- [ ] better-sqlite3
- [ ] cockroachdb
- [ ] cordova
- [ ] expo
- [ ] mongodb
- [ ] mysql
- [ ] nativescript
- [ ] oracle
- [ ] postgres
- [ ] react-native
- [ ] sap
- [ ] spanner
- [ ] sqlite
- [ ] sqlite-abstract
- [ ] sqljs
- [X] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by vinfinity7** [comment]
**Expected Behavior:**
After calling ```repository.save({ timeString: '08:00:00' })```, the database should contain the time value ```'08:00:00'```.

**Actual Behavior:**
When useUTC is set to true, calling ```repository.save({ timeString: '08:00:00' })``` results in incorrect time values, typically` '05:00:00' or '05:00:01'`. When useUTC is set to false, the behavior is inconsistent, and the time values may be off by one second.

**Proposed Solution:**
To ensure consistent and accurate storage of time values in the database, we propose the following solution:

**Step 1: Custom Transformation**
Implement a custom transformation in TypeORM to handle time values consistently. This transformation explicitly converts time values to UTC before storing them in the database.

```
import { Entity, Column, BeforeInsert } from 'typeorm';

@Entity()
export class TimeTestEntity {
  @Column({ type: 'time' })
  timeString!: string;

  @BeforeInsert()
  async convertTimeToUTC() {
    // Ensure that the database connection has useUTC: true
    if (!this.manager.connection.options.useUTC) {
      throw new Error('The database connection must have useUTC set to true.');
    }


    const timeParts = this.timeString.split(':');
    const hours = parseInt(timeParts[0]);
    const minutes = parseInt(timeParts[1]);
    const seconds = parseInt(timeParts[2]);
    const timeInUTC = new Date(Date.UTC(1970, 0, 1, hours, minutes, seconds));

    this.timeString = timeInUTC.toISOString().substr(11, 8);
  }
}```
**Step 2: Configuration**
Ensure that the TypeORM database connection is configured correctly:

useUTC should be set to true in the connection options to ensure consistent handling of UTC time.
**Step 3: Saving Time Values**
When saving time values to the database, use the following approach:
```
const entity = new TimeTestEntity();
entity.timeString = '08:00:00';

const repository = connection.getRepository(TimeTestEntity);
await repository.save(entity);```
**Summary:**
This solution addresses the issue by implementing a custom transformation that converts time values to UTC before storing them in the database. It ensures that the useUTC option is set to true in the database connection configuration for consistent handling of time zones.

By following this solution, we can ensure that time values are consistently and accurately stored in the database, meeting the expected behavior of storing '08:00:00' when calling ```repository.save({ timeString: '08:00:00' }).```

**Comment by penttin5** [comment]
@vifinity7, yes the bug can be circumvented with a custom transformer but  https://github.com/typeorm/typeorm/issues/10397 currently prevents us from using custom transformers, since we need to query the affected column with And-operator

**Comment by vinfinity7** [comment]
@penttin5 , ohk I'll look into this and get back to you
